### PR TITLE
WMP Library v1.0

### PR DESCRIPTION
**Pull request by liangjg**
An attempt was made to re-generate the wmp library using Vector Fitting for all the nuclides. The benefit of this is,  the library is now independent of the resonance parameters and both the data storage and the cross section evaluation can be done in a simpler way. For example, the ``formalism``, ``l_value``, ``pseudo_K0RS`` datasets are removed in the library, the calculation of `sig_t_factor` is also removed in xs evaluation.

Two additional changes made in the new library format:

- in the library generated by VF, the resiudes of scattering xs are independent of energy, so ``residues of total xs`` = ``residues of scattering xs`` + ``residues of absorption xs`` . In the new library, I replaced the residues of total xs with scattering xs so we can get elastic xs residues directly which can be more convenient for elastic scattering treatment.

- the `w_start` and `w_end` arrays which stored the start and end indexes of poles in each window are now combined in one single array `windows`. I think this is clearer and maybe a bit more effecient (more cache friendly).

The library has been validated by comparing with ACE library.

### CODE REVIEW

File: wmp_format.md [hunk 1]
```diff
@@ -1,75 +1,58 @@
...
+
+- **data** (*complex[][]*)
+
+    Complex poles and residues. Each pole has a corresponding set of
+    residues. For example, the i-th pole and corresponding residues
+    are stored as
+
+   ![data](https://latex.codecogs.com/gif.latex?\text{data}[:,i]&space;=&space;[\text{pole}_i,&space;~\text{residue}_{i1},&space;~\text{residue}_{i2},&space;~\text{residue}_{i3}])
+
+    The residues are in the order: scattering, absorption, fission. Complex
```
Comment (paulromano) [inline wmp_format.md]: Once the data is expanded to handle arbitrary reactions, will the format change? It seems like you would need to have something that indicates what reactions poles and residues are stored for.

File: wmp_format.md [hunk 2]
```diff
@@ -1,75 +1,58 @@
...
+    numbers are stored by forming a type with "r" and "i" identifiers,
+    similar to how [h5py] does it.
+
+- **end_E** (*double*)
```
Comment (paulromano) [inline wmp_format.md]: As long as we're changing things, I'd like to propose the following renaming:
- `end_E` --> `E_max`
- `start_E` --> `E_min`

File: wmp_format.md (hunk 1 above)
Comment (liangjg) [reply wmp_format.md]: Yeah, the current format only accounts for 3 fixed reactions. If we extend to include threshold reactions, the format should be more complicated: reactions have different start energies; reactions probably do not have common poles any more in high energy region; reactions may have different window sizes/curve fit order. So I think if the library is expanded, the format will change inevitably. I suggest to keep this version as simple as possible and just use the fixed mts.

File: wmp_format.md (hunk 2 above)
Comment (liangjg) [reply wmp_format.md]: Yes. Will do.

**Comment by liangjg** [comment]
@paulromano Thanks for reviewing. I've already revised openmc  in one branch for this new library. Once this PR is passed, I'll get openmc updated.

**Comment by smharper** [comment]
Thanks for putting this together @liangjg!  I've been looking at the library and everything seems great so far.  I do wonder though for OpenMC purposes if we should add something like a maximum recommended energy cutoff where threshold reactions start occurring.  If I understand correctly, OpenMC won't be able to sample reactions like `(n,2n)` when it is using multipole so we'll get some significant deviations in our results as compared to pointwise.  Maybe we could also do that from the OpenMC side---read both the pointwise data fro thresholds and use that to pick a cutoff for WMP.  What do you think?

**Comment by smharper** [comment]
I wasn't aware `E_max` is limited to occur below the thresholds.  In that case, I see no potential accuracy issues!  I've been trying out this library, and it seems to work great for my purposes.  @paulromano, anything else you want to look at?

**Comment by paulromano** [comment]
Looks good to me too. Thanks @liangjg!

**Pull request merged**